lib/husky_musher/utils/redcap.py

```diff
@@ -0,0 +1,535 @@
+        'instrument': instrument,
+        'event': event,
+        'record': record_id,
+        'returnFormat': 'json'
+    }
+
+    if instance:
+        data['repeat_instance'] = str(instance)
+
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.text
+
+
+def get_todays_repeat_instance() -> int:
+    """
+    Returns the repeat instance number, i.e. days since the start of the study
+    with the first instance starting at 1.
+    """
+    return 1 + (datetime.today() - STUDY_START_DATE).days
+
+
+def redcap_registration_complete(redcap_record: dict) -> bool:
+    """
+    Returns True if a given *redcap_record* shows a participant has completed
+    the enrollment surveys. Otherwise, returns False.
+
+    >>> redcap_registration_complete(None)
+    False
+
+    >>> redcap_registration_complete({})
+    False
+
+    >>> redcap_registration_complete({ \
+        'eligibility_screening_complete': '1', \
+        'consent_form_complete': '2', \
+        'enrollment_questionnaire_complete': '0'})
+    False
+
+    >>> redcap_registration_complete({ \
+        'eligibility_screening_complete': '2', \
+        'consent_form_complete': '2', \
+        'enrollment_questionnaire_complete': '1'})
+    False
+
+    >>> redcap_registration_complete({ \
+        'eligibility_screening_complete': '2', \
+        'consent_form_complete': '2', \
+        'enrollment_questionnaire_complete': '2'})
+    True
+    """
+    if not redcap_record:
+        return False
+
+    return (is_complete('eligibility_screening', redcap_record) and \
+            is_complete('consent_form', redcap_record) and \
+            is_complete('enrollment_questionnaire', redcap_record))
+
+
+def fetch_encounter_events_past_week(redcap_record: dict) -> List[dict]:
+    """
+    Given a *redcap_record*, export the full list of related REDCap instances
+    from the Encounter arm of the project that have occurred in the past week.
+    """
+    # Unfortunately, despite its appearance in the returned response from REDCap,
+    # `redcap_repeat_instance` is not a field we can query by when exporting
+    # REDCap records. Additionally, the `dateRangeBegin` key in REDCap is not
+    # useful to us, because all instances associated with a record are returned,
+    # regardless of the instance's creation or modification date.
+    #
+    # Thus, return all information from a REDCap record (without
+    # specifying fields) to retrieve the instance key.
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'csvDelimiter': '',
+        'events': 'encounter_arm_1',
+        'records': redcap_record["record_id"],
+        'rawOrLabel': 'label',
+        'rawOrLabelHeaders': 'raw',
+        'exportCheckboxLabel': 'false',
+        'exportSurveyFields': 'false',
+        'exportDataAccessGroups': 'false',
+        'returnFormat': 'json'
```

Comment:
After playing with the API a bit more, I found that it always includes the `redcap_repeat_instance` if you include `record_id` in the `fields` filter. So you can specify fields here to reduce the data returned by the API.

Reply:
Sweet, great find. I am using the fields param and requesting `record_id` now